File: app/src/main/java/androidacademy/minsk/lecture3lists/ActorRecyclerActivity.java
package androidacademy.minsk.lecture3lists;

import android.os.Bundle;
import android.widget.Toast;

import androidacademy.minsk.lecture3lists.data.Actor;
import androidacademy.minsk.lecture3lists.data.DataUtil;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;

public class ActorRecyclerActivity extends AppCompatActivity {

    private ActorRecyclerAdapter adapter;

    private final ActorRecyclerAdapter.OnItemClickListener clickListener = position -> {
        Actor actor = adapter.getItem(position);
        String clickMessage = getString(R.string.actor_click_action, actor.getName());

        Toast.makeText(this, clickMessage, Toast.LENGTH_SHORT).show();
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_actor_recycler);

        ActionBar ab = getSupportActionBar();
        if (ab != null) {
            ab.setDisplayHomeAsUpEnabled(true);
        }

        RecyclerView list = findViewById(R.id.actorList);
        adapter = new ActorRecyclerAdapter(this, DataUtil.generateActors(), clickListener);
        list.setAdapter(adapter);
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }
}

File: app/src/main/java/androidacademy/minsk/lecture3lists/ActorRecyclerActivity.kt
package androidacademy.minsk.lecture3lists

import android.os.Bundle
import android.widget.Toast
import androidacademy.minsk.lecture3lists.data.DataUtil.generateActors
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView

class ActorRecyclerActivity : AppCompatActivity() {

    private lateinit var adapter: ActorRecyclerAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_actor_recycler)

        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        val list = findViewById<RecyclerView>(R.id.actorList)
        adapter = ActorRecyclerAdapter(this, generateActors()) {
            val actor = adapter.getItem(it)
            val clickMessage = getString(R.string.actor_click_action, actor.name)
            Toast.makeText(this, clickMessage, Toast.LENGTH_SHORT).show()
        }
        list.adapter = adapter
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }
}